pkg/controller/logstorage/esuserscleanup/ususerscleanup_controller.go

```diff
@@ -0,0 +1,148 @@
+	multiTenant bool
+}
+
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+	if !opts.MultiTenant {
+		// This controller is only meant to run alongside the users controller which is a multi-tenant exclusive controller.
+		return nil
+	}
+
+	r := &CleanupController{
+		client:      mgr.GetClient(),
+		scheme:      mgr.GetScheme(),
+		multiTenant: opts.MultiTenant,
+	}
+
+	// Create a controller using the reconciler and register it with the manager to receive reconcile calls.
+	c, err := controller.New("log-storage-cleanup-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	if err = c.Watch(&source.Kind{Type: &operatorv1.Tenant{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-cleanup-controller failed to watch Tenant resource: %w", err)
+	}
+
+	err = utils.AddPeriodicReconcile(c, utils.PeriodicReconcileTime, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("log-storage-cleanup-controller failed to create periodic reconcile watch: %w", err)
+	}
+
+	return nil
+}
+
+func (r *CleanupController) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	helper := utils.NewNamespaceHelper(r.multiTenant, render.ElasticsearchNamespace, request.Namespace)
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name, "installNS", helper.InstallNamespace(), "truthNS", helper.TruthNamespace())
+	reqLogger.Info("Reconciling LogStorage - Cleanup")
+
+	// Wait for Elasticsearch to be installed and available.
+	elasticsearch, err := utils.GetElasticsearch(ctx, r.client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if elasticsearch == nil || elasticsearch.Status.Phase != esv1.ElasticsearchReadyPhase {
+		return reconcile.Result{}, nil
+	}
+
+	// Clean up any stale users that may have been left behind by a previous tenant
+	if err := r.cleanupStaleUsers(ctx, reqLogger); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *CleanupController) cleanupStaleUsers(ctx context.Context, logger logr.Logger) error {
+	esClient, err := r.esClientFn(r.client, ctx, relasticsearch.ElasticEndpoint())
+	if err != nil {
+		return fmt.Errorf("failed to connect to Elasticsearch - failed to create the Elasticsearch client")
+	}
+
+	allESUsers, err := esClient.GetUsers(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to fetch users from Elasticsearch")
+	}
+
+	tenants := operatorv1.TenantList{}
+	err = r.client.List(ctx, &tenants)
+	if err != nil {
+		return fmt.Errorf("failed to fetch TenantList")
+	}
+
+	clusterIDConfigMap := corev1.ConfigMap{}
+	err = r.client.Get(ctx, client.ObjectKey{Name: "cluster-info", Namespace: "tigera-operator"}, &clusterIDConfigMap)
+	if err != nil {
+		return fmt.Errorf("failed to fetch cluster-info configmap")
+	}
+
+	clusterID, ok := clusterIDConfigMap.Data["cluster-id"]
+	if !ok {
```

Comment:
we also need to check that `clusterID != ""` - the key can exist with an empty string value, which is also no good for us.